src/StravaClient.php

```diff
@@ -23,48 +23,88 @@ public function exchangeToken(string $code, string $grant_type = 'authorization_
         return $this->strava->exchangeToken($code, $grant_type)->json();
     }
 
+    /**
+     * Set access and refresh tokens
+     */
     public function setToken(string $access_token, string $refresh_token): void
     {
         $this->strava->setToken($access_token, $refresh_token);
     }
 
-    public function activityForAthlete($page, $per_page): array
+    /**
+     * Get activities for authenticated athlete
+     *
+     * @throws Exception
+     */
+    public function activityForAthlete(int $page, int $per_page): array
+    {
+        return $this->handleRequest(function () use ($page, $per_page): Response {
+            return $this->strava->activityForAthlete([
+                'page' => $page,
+                'per_page' => $per_page,
+            ]);
+        });
+    }
+
+    /**
+     * Get a specific activity by ID
+     *
+     * @throws Exception
+     */
+    public function getActivity(int $id): array
     {
-        return $this->handleRequest(function () use ($page, $per_page) {
-            return $this->strava->activityForAthlete(['page' => $page, 'per_page' => $per_page]);
+        return $this->handleRequest(function () use ($id): Response {
+            return $this->strava->getActivity($id);
         });
     }
 
+    /**
+     * Handle API request with automatic token refresh
+     *
+     * @throws Exception
+     */
     private function handleRequest(callable $request): array
     {
         $response = $request();
 
-        if ($response->failed()) {
-            return match ($response->status()) {
-                401 => $this->handleUnauthorized($request),
-                404 => throw new Exception('Not Found'),
-                400 => throw new Exception('Bad Request'),
-                default => throw new Exception('Unknown Error'),
-            };
+        if (! $response->failed()) {
+            return $response->json();
         }
 
-        return $response->json();
+        return match ($response->status()) {
+            401 => $this->handleUnauthorized($request),
+            404 => throw new Exception('Resource not found'),
+            400 => throw new Exception($response->json('message') ?? 'Bad request'),
+            429 => throw new Exception('Rate limit exceeded'),
+            500, 502, 503, 504 => throw new Exception('Strava API service error'),
+            default => throw new Exception('Unknown error occurred'),
+        };
```

Comment:
Can you create issues for these exceptions and provide solid recommendations on how to make them useful think of how this might show up in laravel telescope or sentry for example.